Platon/backend/app/follow_system/views.py

```diff
@@ -492,5 +505,118 @@ def delete(user_id, self):
         else:
             return make_response(jsonify({'error': 'Input Format Error'}), 400)
 
+@follow_system_ns.route("/report")
+class ReportAPI(Resource):
+    '''
+        Endpoints for Report Model.
+    '''
+
+    @api.doc(responses={401: 'Account Problems', 400: 'Input Format Error', 500: ' Database Connection Error',
+                        404: 'Not found'})
+    @api.response(200, 'Success', report_list_model)
+    @api.expect(get_report_parser)
+    @login_required
+    @follow_required(param_loc='args', requested_user_id_key='reported_user_id')
```

Comment:
When I imagine a reporting system, I think as if there are some privileged users(such as admins) that can read these reports and take actions. Our current system does not support admin user type, but we can assume admin user type will be created in the near future.

Reply:
However, since this is a course project which contains many assumptions, this is acceptable.